packages/resource-deployment/scripts/install-parallel.sh

```diff
@@ -0,0 +1,145 @@
+Usage: $0 -e <environment> -l <Azure region> -o <organisation name> -p <publisher email> -r <resource group> -s <subscription name or id> -c <client id>
+where:
+Resource group - The name of the resource group that everything will be deployed in.
+Subscription - The subscription for the resource group.
+Environment - The environment in which the set up is running.
+Publisher email - The email for notifications.
+Resource group - The resource group that this API instance needs to be added to.
+Client ID - the app registration client id used for function app authentication/ authorization.
+Azure region - Azure region where the instances will be deployed. Available Azure regions:
+    centralus
+    eastasia
+    southeastasia
+    eastus
+    eastus2
+    westus
+    westus2
+    northcentralus
+    southcentralus
+    westcentralus
+    northeurope
+    westeurope
+    japaneast
+    japanwest
+    brazilsouth
+    australiasoutheast
+    australiaeast
+    westindia
+    southindia
+    centralindia
+    canadacentral
+    canadaeast
+    uksouth
+    ukwest
+    koreacentral
+    koreasouth
+    francecentral
+    southafricanorth
+    uaenorth
+"
+    exit 1
+}
+
+
+function waitForProcesses(){
+    processesToWaitFor=$1
+    for pid in ${processesToWaitFor[@]}; do
+        wait $pid
+        returnCode=$?
+        echo "Value of pid is $pid, Return code is $returnCode"
+    done
+}
+
+# Read script arguments
+while getopts ":r:s:l:e:o:p:c:" option; do
+    case $option in
+    r) resourceGroupName=${OPTARG} ;;
+    s) subscription=${OPTARG} ;;
+    l) location=${OPTARG} ;;
+    e) environment=${OPTARG} ;;
+    o) orgName=${OPTARG} ;;
+    p) publisherEmail=${OPTARG} ;;
+    c) clientId=${OPTARG} ;;
+    *) exitWithUsageInfo ;;
+    esac
+done
+
+if [[ -z $resourceGroupName ]] || [[ -z $subscription ]] || [[ -z $environment ]] || [[ -z $orgName ]] || [[ -z $publisherEmail ]]; then
+    exitWithUsageInfo
+fi
+
+# Login to Azure if required
+if ! az account show 1>/dev/null; then
+    az login
+fi
+
+az account set --subscription "$subscription"
+
+. "${0%/*}/create-resource-group.sh"
+
+. "${0%/*}/create-storage-account.sh"
+
+echo "Finished creating storage account"
+
+resourceGroupSuffix=${storageAccountName:11}
+cosmosAccountName="allycosmos$resourceGroupSuffix"
+apiManagementName="apim-a11y$resourceGroupSuffix"
+
+. "${0%/*}/create-api-management.sh" &
+apiManagmentProcess="$!"
+
+. "${0%/*}/create-datalake-storage-account.sh" &
+parallelizableProcesses="$!"
+
+. "${0%/*}/upload-files.sh" &
+parallelizableProcesses+=" $!"
+
+. "${0%/*}/create-queues.sh" &
+parallelizableProcesses+=" $!"
+
+. "${0%/*}/setup-cosmos-db.sh" &
+parallelizableProcesses+=" $!"
+
+waitForProcesses "${parallelizableProcesses[@]}"
+
+. "${0%/*}/app-insights-create.sh"
+
+. "${0%/*}/batch-account-create.sh"
+
+. "${0%/*}/push-secrets-to-key-vault.sh"
+
+. "${0%/*}/function-app-create.sh"
+
+. "${0%/*}/job-schedule-create.sh"
+
+waitForProcesses "$apiManagmentProcess"
```

Comment:
We are waiting for all processes anyway and could follow same pattern by adding pid into array

Reply:
Some of these need to run sequentially because the predecessor sets some variables that are used in the successor. I haven't found a way to have a background process set the parent's variables. Plus it would be a race condition if we chose to run all in parallel.

Reply:
Can we have a comment of sequences since this is not obvious from the first look or what depends on what?

Reply:
Discussed offline. Adding a comment to indicate dependency.